SampleOfferWithoutReward.sol

```diff
@@ -160,8 +158,7 @@ contract SampleOfferWithoutReward {
 
     // "fire the contractor"
     function returnRemainingEther() onlyClient {
-        if (originalClient.DAOrewardAccount().call.value(this.balance)())
-            isContractValid = false;
+        originalClient.DAOrewardAccount().call.value(this.balance)();
```

Comment:
1. This attack cannot harm you here. It's all transactional, the transfer to the DAO reward account is done or not.
2. Solidity is not affected by this attack.